src/addresses.rs

```diff
@@ -97,38 +163,47 @@ impl Default for ChainAddresses {
     }
 }
 
-pub fn mainnet_address() -> ChainAddresses {
-    ChainAddresses {
-        v1_mixed_route_quoter_address: Some(address!("84E44095eeBfEC7793Cd7d5b57B7e401D7f1cA2E")),
-        ..Default::default()
-    }
+lazy_static! {
+    pub static ref MAINNET_ADDRESS: ChainAddresses = {
+        ChainAddresses {
+            v1_mixed_route_quoter_address: Some(address!(
+                "84E44095eeBfEC7793Cd7d5b57B7e401D7f1cA2E"
+            )),
+            ..Default::default()
+        }
+    };
 }
 
-pub fn goerli_address() -> ChainAddresses {
-    ChainAddresses {
-        v1_mixed_route_quoter_address: Some(address!("Ba60b6e6fF25488308789E6e0A65D838be34194e")),
-        ..Default::default()
-    }
+lazy_static! {
+    pub static ref GOERLI_ADDRESS: ChainAddresses = {
+        ChainAddresses {
+            v1_mixed_route_quoter_address: Some(address!(
+                "Ba60b6e6fF25488308789E6e0A65D838be34194e"
+            )),
+            ..Default::default()
+        }
+    };
 }
 
-pub fn optimism_addresses() -> ChainAddresses {
-    ChainAddresses::default()
+lazy_static! {
+    pub static ref OPTIMISM_ADDRESSES: ChainAddresses = ChainAddresses::default();
 }
 
-pub fn arbitum_one_addresses() -> ChainAddresses {
-    ChainAddresses {
-        multicall_address: address!("adF885960B47eA2CD9B55E6DAc6B42b7Cb2806dB"),
-        tick_lens_address: Some(address!("bfd8137f7d1516D3ea5cA83523914859ec47F573")),
-        ..Default::default()
-    }
+lazy_static! {
+    pub static ref ARBITUM_ONE_ADDRESSES: ChainAddresses = {
+        ChainAddresses {
+            multicall_address: address!("adF885960B47eA2CD9B55E6DAc6B42b7Cb2806dB"),
+            tick_lens_address: Some(address!("bfd8137f7d1516D3ea5cA83523914859ec47F573")),
+            ..Default::default()
+        }
+    };
 }
-
-pub fn polygon_addresses() -> ChainAddresses {
-    ChainAddresses::default()
+lazy_static! {
+    pub static ref POLYGON_ADDRESSES: ChainAddresses = ChainAddresses::default();
 }
 
 /// celo v3 addresses
-pub fn celo_addresses() -> ChainAddresses {
+pub const CELO_ADDRESSES: ChainAddresses = {
```

Comment:
Redundant `{`.

Reply:
true, but my rust-analyzer has this red stuff near the closing bracket